fix(website): prevent sidebar scrolling from jumping

peduarte opened the pull request:
An attempt at trying to persist the previous sidebar scroll position.

Not quite happy with this approach. I think it would be better if the sidebar didn't re-render at all.

It works... But does anyone have a better solution?

---

Anything inside `wrapRootElement` doesn't get unmounted on page change, so I tried moving the `StaticQuery` into `wrapRootElement` but it doesn't work over there :/

hadihallak commented on packages/website/src/components/DocLayout.js:
```diff
@@ -121,3 +57,87 @@ function DocLayout({ children, pathname }) {
+  const sidebarRef = useRef(null);
+  const { current, setPrevious } = useContext(SidebarScrollContext);
+
+  useEffect(() => {
```
using `useLayoutEffect` here would be better as it makes sure that the user will not see the scroll before it gets updated to a new scrolling position.

peduarte commented:
**UPDATE**

I've managed to find a different way to solve the issue.

- replaced `wrapRootElement` with `wrapPageElement: turns out `StaticQuery` works there
- removed `DocLayout`


For now this is now an issue, since we display the sidebar on all pages. But eventually when we have a homepage for Radix, I'll probably need to check `props.location` in `wrapPageElement` to conditionally render the Sidebar.

It's not a perfect solution and I'm surprised there isn't an easier way to do this with Gatsby. I checked other websites such as [reactjs](http://reactjs.org) and [styled-system](https://styled-system.com/#getting-started) but none of them seem to have fixed the issue.
